common/src/Assets/EntityModel.cpp

```diff
@@ -500,12 +506,9 @@ std::unique_ptr<Renderer::TexturedRenderer> EntityModel::buildRenderer(
   const auto actualSkinIndex = skinIndex + frame->skinOffset();
   for (const auto& surface : m_surfaces)
   {
-    if (actualSkinIndex < surface->skinCount())
+    if (auto renderer = surface->buildRenderer(actualSkinIndex, frameIndex))
```

Comment:
This means that we will build a new renderer for every invalid skin index we get here. Would it not be better to correct the skin index here and reuse the renderer for the default index 0?

Reply:
I was wondering what the better option would be between the two, initially I had it that way but thought this looked cleaner. But you're right, that way would use fewer resources.